ql/src/java/org/apache/hadoop/hive/ql/parse/TaskCompiler.java

```diff
@@ -517,17 +517,21 @@ private Path getDefaultCtasLocation(final ParseContext pCtx) throws SemanticExce
     try {
       String protoName = null, suffix = "";
       boolean isExternal = false;
-      
+      boolean useSuffix = HiveConf.getBoolVar(conf, HiveConf.ConfVars.HIVE_ACID_CREATE_TABLE_USE_SUFFIX)
+              || HiveConf.getBoolVar(conf, HiveConf.ConfVars.HIVE_ACID_LOCKLESS_READS_ENABLED);
+
       if (pCtx.getQueryProperties().isCTAS()) {
         protoName = pCtx.getCreateTable().getDbTableName();
         isExternal = pCtx.getCreateTable().isExternal();
-      
+        if (!isExternal && useSuffix) {
+          long txnId = Optional.ofNullable(pCtx.getContext())
+                  .map(ctx -> ctx.getHiveTxnManager().getCurrentTxnId()).orElse(0L);
+          suffix = SOFT_DELETE_PATH_SUFFIX + String.format(DELTA_DIGITS, txnId);
+        }
       } else if (pCtx.getQueryProperties().isMaterializedView()) {
         protoName = pCtx.getCreateViewDesc().getViewName();
-        boolean createMVUseSuffix = HiveConf.getBoolVar(conf, HiveConf.ConfVars.HIVE_ACID_CREATE_TABLE_USE_SUFFIX)
-          || HiveConf.getBoolVar(conf, HiveConf.ConfVars.HIVE_ACID_LOCKLESS_READS_ENABLED);
 
-        if (createMVUseSuffix) {
+        if (useSuffix) {
```

Comment:
createTableUseSuffix &=
 AcidUtils.isTablePropertyTransactional(pCtx.getCreateViewDesc().getTblProps());

Reply:
Should we do transactional check for Materialised views?

Reply:
Since suffixing must be done for transactional MV, added a condition. Updated.